Feature mp

mlohvynenko opened the pull request.

mlohvynenko committed:
conan: add packages needed by message proxy

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:conan: remove redundant conan files

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
host_build: add mp flags

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: remove reduntant host build script

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: remove git submodules and CI files

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: remove format files

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: remove redundant cmake files

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
cmake: add with vchan option

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: organize files according to mono repo template

- headers are located with sources;
- each module has `tests` folder inside with its unit tests;
- '_test' suffix remove from tests source files.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
cmake: extend add_target with properties option

This patch extends add_target cmake function with
properties option. This allows to set test target
properties for unit tests like environment variables, needed
by test cases that work with custom softhsm and
openssl configuration files. As well as target properties.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
common:version: automatically generate version header file

This patch add cmake script and template file
to automatically generate version.hpp file
based on git tags and commit hash.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
cmake: add iam, mp and sm project options

This patch adds cmake options for iam, mp and sm
targets and enables aos protocol options based
on these new options.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:cmake: use add module cmake functions

This patch update message proxy cmake files to
use functions from add module cmake file to
declare targets.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: update includes according to mono repo pattern

- local to module includes should be last and without path;
- other modules includes should contain full path,
  for example: 'common/utils/exception.hpp'.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: change header guards naming

Change all defs in header guard to match the following format:
AOS_{COMPONENT}_{MODULE}_{FILENAME}_HPP_

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: remove redundant version header template

This patch removes the redundant version header
template from the message proxy module, as global
version header is already included in the project.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp: remove outdated readme file

This patch removes readme file for the mp module,
as it contains outdated information and is no longer
relevant to the current state of the project.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

al1img commented on CMakeLists.txt:
```diff
@@ -18,6 +18,7 @@ set(AOS_CORE_DIR
     CACHE STRING "path where Aos core lib, core API and other dependencies are located"
 )
 
+option(WITH_VCHAN "build with vchan" ON)
```
Add empty line after, let's divide test/cov/doc with functionality flags

al1img commented on src/mp/version.hpp.in:
As agreed, let's create interface target under common folder

al1img commented on cmake/AddModule.cmake:
```diff
@@ -12,6 +12,7 @@
 #     SOURCES <list>         - list of source files;
 #     LINK_OPTIONS <list>    - list of link options (optional);
 #     LIBRARIES <list>       - list of libraries to link against (optional).
+#     PROPERTIES <list>      - list of properties for test targets (optional).
```
Then it should named as TEST_PROPERTIES

or

it could be more generic and used for `set_target_properties` command.

Could you check if `set_target_properties`  can be used instead of `gtest_discover_tests` with properties?

mlohvynenko replied:
Looks like set_target_properties applicable only for build process, when properties arg for gtest_discover_tests affects built executable runtime
renamed to TEST_PROPERTIES

mlohvynenko committed:
mp:app: move app into aos::mp::app namespace

This patch moves the app class into the aos::mp::app namespace
and removes redundant aos:: namespace usage.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:cmclient:tests: remove redundant aos namespace usage

This patch removes the redundant usage of the aos
and moves static functions to an unnamed namespace.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:communication: remove redundant aos namespace usage and comments

This patch remove redundant usage of the aos namespace
and cleans up commented code.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:config:tests: remove redundant aos namespace usage

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:filechunker:tests: remove redundant aos namespace usage

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:iamclient:tests: remove redundant aos namespace usage

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
mp:imageunpacker:tests: remove redundant aos namespace usage

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mykola-kobets-epam commented on src/mp/app/CMakeLists.txt:
```diff
@@ -0,0 +1,52 @@
+    Poco::Util
+    aoscommon
+    aoscrypto
+    aos::common::downloader
```
I used COMMON_NAMESPACE alias from MSolyanko PR

mlohvynenko committed:
cmake: move FindXenvchan.cmake to the root cmake directory

This patch moves the FindXenvchan.cmake from mp/cmake to the root
cmake directory, to not update CMAKE_MODULE_PATH in mp directory.

Signed-off-by: Mykhailo Lohvynenko <Mykhailo_Lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mykola-kobets-epam commented:
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`
